make/jdk/src/classes/build/tools/taglet/ToolGuide.java

```diff
@@ -151,6 +151,12 @@ private String docRoot(Element elem) {
                 return pe.getEnclosingElement() != null
                         ? "../" + pkgPart
                         : pkgPart;
+            case CLASS:
+                TypeElement te = (TypeElement)elem;
+                return te.getQualifiedName()
+                        .toString()
+                        .replace('.', '/')
+                        .replaceAll("[^/]+", "..");
```

Comment:
If the class is in a module don't you have to get one step higher to get the root?
I am not familiar with this code, so I'm just reasoning by induction here - trying to match with what the case for PACKAGE seems to be doing...

Reply:
Same here, I initially applied the same pattern as for PACKAGE but that does not produce the correct path (it includes 1 ".." too much.):
```
                String typePart = te.getQualifiedName()
                        .toString()
                        .replace('.', '/')
                        .replaceAll("[^/]+", "..");
                return te.getEnclosingElement().getEnclosingElement() != null
                        ? "../" + typePart
                        : typePart;
```

Reply:
I see - class has a class name which is a file - so `foo.Bar` produces `../..` relative to the directory foo (foo/../..) - so it already has an additional `..` compared to package - where `a.b` would produce ../.. relative to *directory* b which would lead to `b/../..` and not `a/../..` which is what we need.